builtin/rebase.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * "git rebase" builtin command
+ *
+ * Copyright (c) 2018 Pratik Karki
+ */
+
+#include "builtin.h"
+#include "run-command.h"
+#include "exec-cmd.h"
+#include "parse-options.h"
+#include "argv-array.h"
+#include "refs.h"
+#include "dir.h"
+
+enum rebase_type {
+	PRESERVE_MERGES
+};
+
+static GIT_PATH_FUNC(git_path_rebase_apply, "rebase-apply");
+static GIT_PATH_FUNC(git_path_rebase_merge, "rebase-merge");
+
+enum rebase_type mode;
+
+static int use_builtin_rebase(void)
+{
+	struct child_process cp = CHILD_PROCESS_INIT;
+	struct strbuf out = STRBUF_INIT;
+	int ret;
+
+	argv_array_pushl(&cp.args,
+			 "config", "--bool", "rebase.usebuiltin", NULL);
+	cp.git_cmd = 1;
+	if (capture_command(&cp, &out, 6))
+		return 0;
+
+	strbuf_trim(&out);
+	ret = !strcmp("true", out.buf);
+	strbuf_release(&out);
+	return ret;
+}
+static int finish_rebase(void)
+{
+	const struct object_id *new_head;
+	struct strbuf reflog_message = STRBUF_INIT;
+	struct commit *head_commit;
+	const struct object_id *head = &head_commit->object.oid;
+	unlink("$(git rev-parse --git-path REBASE_HEAD)"); //NEEDS HELP.
+	//apply_autostash();
+	const char *argv_gc_auto[] = { "gc", "--auto", NULL };
```

Comment:
This is a declaration after a statement (the unlink() call). We try to avoid those.

Reply:
Compile with `DEVELOPER=1` to catch such issues.